src/v/cluster/self_test_frontend.cc

```diff
@@ -43,6 +43,13 @@ self_test_status self_test_frontend::node_test_state::status() const {
     return response->status;
 }
 
+self_test_stage self_test_frontend::node_test_state::stage() const {
+    if (!response) {
+        return self_test_stage::idle;
+    }
+    return response->stage;
+}
```

Comment:
> rpk: more verbose self-test status report

there is a lot of changes to c++/non-rpk in this commit, but the commit describes changes to rpk. i think these should be split out into a separate (or folded into a previous) commit where the logical changes are made and described.

Reply:
Split the commit.